cpu/native/fs/native_fs.c

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (C) 2023 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_fs_native
+ * @{
+ *
+ * @file
+ * @brief       native integration with vfs
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <string.h>
+
+#include "mutex.h"
+#include "native_internal.h"
+#include "fs/native_fs.h"
+#include "vfs.h"
+
+#define ENABLE_DEBUG 0
+#include <debug.h>
+
+#define FD(filep) filp->private_data.value
+#define DIRP(dirp) dirp->private_data.ptr
+
+static mutex_t _lock;
+
+static void _do_prefix(vfs_mount_t *mountp, const char *name, char *buffer, size_t len)
+{
+    const native_desc_t *fs_desc = mountp->private_data;
+
+    getcwd(buffer, len);
+
+    len = sizeof(buffer) - strlen(buffer);
+    strncat(buffer, fs_desc->hostpath, len);
+
+    len = sizeof(buffer) - strlen(buffer);
+    strncat(buffer, name, len);
+}
+
+static char *_prefix_path(vfs_mount_t *mountp, const char *name)
+{
+    static char buffer[PATH_MAX];
+
+    _do_prefix(mountp, name, buffer, sizeof(buffer));
+
+    return buffer;
+}
+
+static int _mount(vfs_mount_t *mountp)
+{
+    int res;
+    mutex_lock(&_lock);
+
+    real_mkdir(_prefix_path(mountp, ""), 0755);
```

Comment:
Why not rely on the umask?

Reply:
Do you mean setting it to `0`? That just gives

```
d--------- 1 benpicco benpicco       0 Feb 25 02:27 nvm0
```

Reply:
No, setting mode to 0o777. Depending on the user's umask (typically 0o022 or 0o027) the directory will then have 0o755 or 0o750.

Reply:
Ah yes, that works